### PR TITLE
AZP/SNAPSHOT: Fix snapshot Pipeline

### PR DESCRIPTION
**Pull request by avildema**
## Why ?
Snapshot pipeline skip now . condition for launch is null

## How ?
Add conditional for launch in snapshot pipeline

### CODE REVIEW

File: buildlib/azure-pipelines.yml [hunk 1]
```diff
@@ -1,20 +1,53 @@
...
+              then
+                  run_by_title=Yes
+              fi
+
+              if [[ ( "$(Build.Reason)" == "IndividualCI" ) || ( "${run_by_title}" == "Yes" && "$(Build.Reason)" == "PullRequest" ) ]]
+              then
+                  echo "##vso[task.setvariable variable=Launch;isOutput=true]Yes"
+              else
+                  echo "##vso[task.setvariable variable=Launch;isOutput=true]No"
+              fi
```
Comment (yosefe) [inline buildlib/azure-pipelines.yml:40]: can we put this code in some common place for {snapshot,release,pr}.yml to use?

File: buildlib/azure-pipelines.yml [hunk 2]
```diff
@@ -1,20 +1,53 @@
...
 # This pipeline to be run on direct pushes and merges
 
-pr: none
 trigger:
   - master
   - v*.*.x
 
+pr:
+  - master
+  - v*.*.x
```
Comment (yosefe) [inline buildlib/azure-pipelines.yml:10]: this is temp?

Comment (avildema) [reply buildlib/azure-pipelines.yml:10]: For check pipeline on PR, can remove or or not
We check it only if PR have prefix `AZP/SNAPSHOT: `

File: buildlib/az-helpers.sh [hunk 3]
```diff
@@ -130,3 +130,24 @@ try_load_cuda_env() {
...
+    if [[ "$title" == "$title_mask"* ]]
+    then
+      run_by_title=Yes
+    fi
+
+    if [[ ( "$build_reason" == "IndividualCI" ) || ( "${run_by_title}" == "Yes" && "$build_reason" == "PullRequest" ) ]]
+    then
+      echo "##vso[task.setvariable variable=Launch;isOutput=true]Yes"
+    else
+      echo "##vso[task.setvariable variable=Launch;isOutput=true]No"
```
Comment (yosefe) [inline buildlib/az-helpers.sh:151]: minor: indent by 4 spaces, not 2

File: buildlib/az-helpers.sh [hunk 4]
```diff
@@ -130,3 +130,24 @@ try_load_cuda_env() {
...
+    build_reason=$3
+    echo "Get commit message target $git_id"
+    title=`git log -1 --format="%s" $git_id`
+    run_by_title=No
+    if [[ "$title" == "$title_mask"* ]]
+    then
+      run_by_title=Yes
+    fi
+
+    if [[ ( "$build_reason" == "IndividualCI" ) || ( "${run_by_title}" == "Yes" && "$build_reason" == "PullRequest" ) ]]
```
Comment (yosefe) [inline buildlib/az-helpers.sh:147]: maybe replace `"${run_by_title}" == "Yes"` by `"$title" == "$title_mask"*`

**Review by yosefe — COMMENTED** [review]
can you pls update why/how PR description

**Review by yosefe — APPROVED** [review]

**Comment by yosefe** [comment]
@avildema can u pls squash?

**Review by petro-rudenko — APPROVED** [review]

**Comment by petro-rudenko** [comment]
@avildema  still failed to publish snapshot:
https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=16092&view=logs&j=fd52c9e6-58ae-5de8-d2ba-8344c0a0876d&t=4a735699-dab6-5720-7942-0366dbf89ff2

`/home/vsts/work/_temp/3cd5fd8c-2894-4f99-b292-011e3402f955.sh: line 4: system.pullRequest.sourceCommitId: command not found`

**Pull request merged**